pkg/metrics/exporter.go

```diff
@@ -40,52 +33,22 @@ func (r *runner) Start(ctx context.Context) error {
 	log.Info("Starting metrics runner")
 	defer log.Info("Stopping metrics runner workers")
 	errCh := make(chan error)
-	go func() { errCh <- r.newMetricsExporter() }()
+	exporters := registry.Exporters()
+	for i := range exporters {
+		startExporter := exporters[i]
+		go func() {
+			if err := startExporter(ctx); err != nil {
+				errCh <- err
```

Comment:
If we have 2+ active exporters and one of them failed to start, we will terminate immediately (based on the select statement below). Is that the desired behavior?

Reply:
Correct, that is the behavior controller-manager uses. This is to make sure users are aware that there is a running pod that is not serving metrics (otherwise it'd be a silent failure).

If, for some reason, silent failure is desirable, then the exporter can swallow the error, like the stackdriver exporter does.